dbms/programs/server/TCPHandler.cpp

```diff
@@ -170,8 +171,9 @@ void TCPHandler::runImpl()
             send_exception_with_stack_trace = query_context->getSettingsRef().calculate_text_stack_trace;
 
             /// Should we send internal logs to client?
+            CLIENT_LOGS_LEVEL = query_context->getSettingsRef().send_logs_level.value;
```

Comment:
Race condition when multiple concurrent queries are run: each will overwrite logs level.

Reply:
Look at ThreadsStatus, ThreadStatusExt.

Reply:
Race condition on atomic variable?

Reply:
Yes.

Reply:
Now you have to make .sh test that runs two queries in parallel.